js/navbar/tabColor.js

```diff
@@ -243,7 +200,61 @@ function setColor (bg, fg) {
   }
 }
 
-// theme changes can affect the tab colors
-window.addEventListener('themechange', function (e) {
-  updateColorPalette()
-})
+const tabColor = {
+  initialize: function () {
+    webviews.bindEvent('page-favicon-updated', function (e, favicons) {
```

Comment:
You moved this method here probably with the hope to make tabColor.js easier to maintain, but in fact it makes it harder because it couples tabColor to webviews.

webviews.js binds all kinds of events and I think it makes sense if it serves some sort of an initializator for other modules. It'll be cleaner if they stay webviews-independent, and instead webviews loads them and initializes them.

But, as I don't have such a good picture in my mind of the entire browser, maybe you have strong reasons to think that's not the case. In that situation, to avoid doing this funny deferred loading, you can take a "webviews" parameter from `tabColor::initialize()`. Then, in the webviews, you'd call `tabColor.initialize(webviews)`. This way, `tabColor` is independent of the global `webviews` and you keep yourself from doing strange deferred loadings :)

Reply:
> Interesting, I was actually planning in doing the opposite (moving all the events out of webviews.js and into the modules that they relate to.

I support that. I wouldn't have mentioned it at all if I knew you had these intentions.

> Keeping the events in the module does make it a bit easier to understand how the module works, and I was thinking it should allow us to add and remove modules more easily than if everything was linked together in webviews.js

You're right. As I didn't know you plan to apply it to everything, it seemed strange to me to export some 10 lines out of a 500 LOC file, but it makes perfect sense to continually reduce the webviews.js to something more easy to work with.

:+1:

Reply:
> maybe it is better to do what you're suggesting, and just have the webview decide which modules to notify based on the events that are happening.

Are you talking about receiving `webviews` as a dependency of `initialize()`? Or about not having `initialize()` at all?

We can have an `initialize(webviews)` per module, and call it inside `webviews.js`.

Reply:
No, I just meant leaving the events inside the webviews.js file - sorry that wasn't worded very well!

Eventually, we should just be able to have each module use `const webviews = require("webviews.js")`, but since the webviews aren't a module yet, we can't do that.

Reply:
In that case, you can just bring `tabColor::initialize()` back to `js/webviews.js` (with the theme change listener) and remove `deferredLoad.js` to resolve this (last) conversation.

Reply:
Assuming that tabColor is a completely separate module with all the webview events in it, I think the eventual goal would be to initialize it from `default.js` and have the tabColor module import the webview module. I can move the `require("tabColor.js")` into webviews.js, but I don't really see why that would be any better than leaving it separate.